### ISSUE TITLE
Cannot use in Windows Phone 8.1 project.

### ISSUE THREAD

**Issue by moonClimber** [luberda-molinet/FFImageLoading#424]
I try to use FFImageLoading in a WP8.1 project but, even after many many attempts and different approaches I always have the following error:
"**The application called an interface that was marshalled for a different thread. (Exception from HRESULT: 0x8001010E (RPC_E_WRONG_THREAD))**" 

By running the sample provided with the source code, **Simple.WinPhone.Sample (Windows Phone 8.1)** I can obtain the same error, that occurs in **line 302** of **file WorkScheduler.cs** in **FFImageLoading.Work.WorkScheduler namespace** (**FFImageLoading (Portable) project**).

Just run the code (both in emulator and device) and wait for a while and after few seconds, without any interaction, the exception raises.

Any suggestions?
Thanks

Here is the **stack**

at Windows.UI.Xaml.DependencyObject.GetValue(DependencyProperty dp)
   at FFImageLoading.FFImage.get_SuccessCommand()
   at FFImageLoading.FFImage.OnSuccess(SuccessEventArgs e)
   at FFImageLoading.FFImage.<LoadImage>b__12_2(ImageInformation imageInformation, LoadingResult loadingResult)
   at FFImageLoading.Work.WorkScheduler.<>c__DisplayClass47_0.<CreateFrameworkTask>b__0(ImageInformation size, LoadingResult result)
   at FFImageLoading.Work.ImageLoaderTask`2.<RunAsync>d__109.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at FFImageLoading.Work.WorkScheduler.<RunImageLoadingTaskAsync>d__50.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at FFImageLoading.Work.WorkScheduler.<RunImageLoadingTaskAsync>d__50.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at FFImageLoading.Work.WorkScheduler.<TakeFromPendingTasksAndRunAsync>d__49.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable.ConfiguredTaskAwaiter.GetResult()
   at FFImageLoading.Work.WorkScheduler.<TakeFromPendingTasksAndRun>d__46.MoveNext()

**Comment by daniel-luberda** [comment]
It's related to this: #166 - there's a solution there. Please see it.

**Comment by moonClimber** [comment]
I've tried to apply the solution in #166, but it still does not work, and I suspect the problem could be different and not a duplicate.

Here are some notes from my attempts I hope can be useful to understand the problem better.
First of all: the problem is replicable by executing the project Simple.WinPhone.Sample (Windows Phone 8.1) ... I'm using both **Device** (_Lumia 930_) and "**Mobile Emulator 10.0.10586.0 WVGA 4 inch 512MB**"

If I change the code as suggested in #166 an exception raises immediately in MainThreadDispatcher.Post.
I've tried to get the dispatcher with different statements such as:

- `CoreApplication.GetCurrentView().Dispatcher`
- `CoreWindow.GetForCurrentThread().Dispatcher`

So, I MUST use your original statement "`CoreApplication.MainView.CoreWindow.Dispatcher`" to get the dispatcher and to make the MainThreadDispatcher.Post working: 

But, you're maybe wondering: "where is the problem if with my original code the method work"?
I think that the problem is somewhere else.

I suppose that there should be at least three logic main steps in the app:

- **STEP1**: show the image placeholder (the loading image)
- **STEP2**: show the actual image (this file http://loremflickr.com/600/600/nature?filename=simple.jpg from MainPage.xaml)
- **STEP3**: something other operation (maybe related to saving file in cache or something else)

So, if I apply the suggestion in #166 the exception occurs before the STEP1. 
With the original code, the exception occurs in STEP3.
In other words: 

- with the #166 solution, the View doesn't start

- with the original code, I can see the final image (loaded asynchronously) and after a little while there is the error.

So, anyone can replicate the problem just by running the sample code?
Can anyone help me to solve this issue? 

Thanks

**Comment by daniel-luberda** [comment]
@moonClimber Could you test the newest prerelease nuget? There were some fixes that could potentially fix your issues.

**Comment by moonClimber** [comment]
No way to solve the problem with the original code. I've pulled a couple of change proposals (one fix and one improvement).
Hope you want to merge them.
